.github/workflows/build_and_test.yml

```diff
@@ -458,8 +457,17 @@ jobs:
       - name: Configure ECR login
         run: echo "{\"credsStore\":\"ecr-login\"}" > /kaniko/.docker/config.json
 
-      - name: Kaniko build compute node with extensions
-        run: /kaniko/executor --skip-unused-stages --snapshotMode=redo --cache=true --cache-repo 369495373322.dkr.ecr.eu-central-1.amazonaws.com/cache --snapshotMode=redo --context . --dockerfile Dockerfile.compute-node --destination 369495373322.dkr.ecr.eu-central-1.amazonaws.com/compute-node:$GITHUB_RUN_ID
+        # compute-node uses postgres 14, which is default now
+        # cloud repo depends on this image name, thus duplicating it
+        # remove compute-node when cloud repo is updated
+      - name: Kaniko build compute node with extensions v14 (compatibility)
+        run: /kaniko/executor --skip-unused-stages --snapshotMode=redo --cache=true --cache-repo 369495373322.dkr.ecr.eu-central-1.amazonaws.com/cache --snapshotMode=redo --context . --dockerfile Dockerfile.compute-node-v14 --destination 369495373322.dkr.ecr.eu-central-1.amazonaws.com/compute-node:$GITHUB_RUN_ID
+        # Note: don't use cache for v14 and v15 images, because they are built from different Dockerfiles
```

Comment:
I've tried to fix this failure:
https://github.com/neondatabase/neon/runs/8186945974?check_suite_focus=true

IIUC, that's because internally these images use non-versioned paths (i.e. usr/local/bin) and when they are cached in the same `cache-repo`, it becomes a mess.

Probably the better fix would be to just use different cache repos.

Reply:
Hmm. 
I thought that Kaniko uses different containers for different steps (`run` commands), but checks the cache (if enabled) to reuse the artifacts between steps.

Reply:
Apparently, that isn't true.
@zoete , can you help me to fix this?
The final goal here is to build two compute images (v14 and v15). And also keep compat image for a transition period.

Should I use `--cleanup` option to cleanup the filesystem between builds, or there is another solution?

Reply:
I've added `cleanup`, but the error persists.
```
error building image: could not save file: copying file: symlink postgres /kaniko/1/usr/local/pgsql/bin/postmaster: file exists
```
Is it a kaniko bug, or I am missing something?

Reply:
Make each `run: /kaniko/executor` a separate job (so we get compute-node-image-14, 15 etc) as I am not 100% sure the kaniko cleanup does the trick in combination with github actions logic.